src/rgw/rgw_file.h

```diff
@@ -510,7 +516,12 @@ namespace rgw {
     int readdir(rgw_readdir_cb rcb, void *cb_arg, uint64_t *offset, bool *eof,
 		uint32_t flags);
     int write(uint64_t off, size_t len, size_t *nbytes, void *buffer);
-    int write_finish();
+
+    int commit(uint64_t offset, uint64_t length, uint32_t flags) {
+      return 0;
+    }
```

Comment:
I wouldn't. If you give NFS3ERR_NOTSUPP in response to COMMIT you'll make the kernel freak out and probably record an item in non-volatile memory to remind itself not to talk to that server again.